### ISSUE TITLE
Create a page of current structs that are stubs and identifying how they can be fully integrated into the game

### ISSUE THREAD

**Issue by nicholaswlee** [uchicago-cs/chiventure#1232]
We will examine all of the structs currently contained within the battle module to identify which are stubs and how these stubs can be fully implemented and integrated with other modules. The product will be a design document on the wiki detailing our findings.

Members: Nicholas, Nafis, Fatimah

**Comment by nicholaswlee** [comment]
Fatimah, Nafis, and I have met and compiled a[ document](https://github.com/uchicago-cs/chiventure/wiki/Battles-~-Current-Battle-Structs-2022) with all of the structs that exist within the Battles module.

**Comment by corian-jal** [comment]
After examining all of the structs in the battle module, we found a lot of similarly defined structs in game_state's stat.h. We have concluded that a previous battle team implemented them to facilitate their own work while waiting for game-state's formal implementation, and we are now considering which to use and how to integrate them. We are also looking around to see if other teams have implemented other structs we have stubs for.

**Comment by corian-jal** [comment]
In npc_battle.h, a struct for NPC battle combatants is defined to contain health, stats, and other similarities with our combatant struct but along with fields we do not have such as hostility and surrender levels. We plan to speak to the NPC team about merging their implementation of npc_battle_t with our combatant struct.

**Comment by corian-jal** [comment]
In skilltrees_common.h, skill_t is defined to take an effect_t, which is forward declared for the player class module. We have determined that effect refers to the effect of a skill (offensive buff, defensive buff, resistances, etc) rather than status effects. As such, we will leave it to those teams and wait for their lead in implementing an if-condition/then-effect function in the battle module.

**Comment by corian-jal** [comment]
Considering the similarities between skills and status effects (increases v. decreases a state, stopping attacks of others v. the player, etc) we are thinking about leaving the implementation behind status effects to the Skill Tree team and either borrowing it for status effects or giving them prerogative over status effects as negative skills.

**Comment by corian-jal** [comment]
The player class struct in class_structs.h has a field of type stats_hash_t and we would like to make sure that they either meant to use it or update them to our stats struct.

**Comment by nmkhan100** [comment]
Here is a link to the page we worked on: (https://github.com/uchicago-cs/chiventure/wiki/Battles-~-Current-Battle-Structs-2022)

We looked through every module to find structures that are relevant to our battle module. We have located a number of these structures in the game state module and one of them in the NPC module. We have also provided a short description of the purpose of these structures and the parts of these structures that are relevant to our battle module. Most importantly, we talked about how we would integrate our data structures with these structures that we have located, whether it requires us to fix and create new data types on our side, or for other teams to do this in order to fix any future conflicts.
